docs/snapshotting/versioning.md

```diff
@@ -40,41 +60,51 @@ A microVM state file is further split into four different fields:
+Firecracker reads and writes the `state` blob of the snapshot by using per version, separate serialization and deserialization logic. This logic is mostly autogenerated by a Rust procedural macro based on `struct` and `enum` annotations. Basically, one can say that these structures support versioning. The versioning logic is generated by parsing a structure's history log (encoded using Rust annotations) and emitting Rust code.
 
 Versioned serialization and deserialization is divided into two translation layers:
- - structural translator,
+ - field translator,
  - semantic translator.
 
-The _structural translator_ implements the logic to convert between different versions of the same Rust POD structure: it can deserialize or serialize from source version to target. 
+The _field translator_ implements the logic to convert between different versions of the same Rust POD structure: it can deserialize or serialize from source version to target. 
 The translation is done field by field - the common fields are copied from source to target, and the fields that are unique to the target are (de)serialized with their default values.
 
-The _semantic translator_ works on top of the structural one and its only concern is translating the semantics of the serialized/deserialized fields. 
+The _semantic translator_ is only concerned in translating the semantics of the serialized/deserialized fields. 
 
-The _structural translator_ is generated automatically through a procedural macro, and the _semantic translation methods_ have to be annotated in the structure by the user.
+The _field translator_ is generated automatically through a procedural macro, and the _semantic translation methods_ have to be annotated in the structure by the user.
 
 This block diagram illustrates the concept:
 
 ![Versionize](
 ../images/versionize.png?raw=true
 "Versionize layers")
 
-## Bincode encoding
-During research and prototyping we have considered multiple storage formats. The criteria used for comparing these were performance, size, Rust support, specification, versioning support, community popularity and tooling. Performance, size and Rust support are hard requirements while all others can be the subject of trade-offs. 
+## VM state encoding
+
+During research and prototyping we considered multiple storage formats. The criteria used for comparing these are: performance, size, rust support, specification, versioning support, community and tooling. Performance, size and Rust support are hard requirements while all others can be the subject of trade offs.
+More info about this comparison can be found here (https://github.com/firecracker-microvm/firecracker/blob/9d427b33d989c3225d874210f6c2849465941dc0/docs/snapshotting/design.md).
+
+Key benefits of using *bincode* versus everything else:
```

Comment:
versus everything else we've tried and tested

Reply:
That is implied in the link above. I will rephrase to make it sound better.